usecases/objects/batch_add.go

```diff
@@ -164,21 +166,33 @@ func (b *BatchManager) validateObject(ctx context.Context, principal *models.Pri
 	if _, ok := fieldsToKeep["lastUpdateTimeUnix"]; ok {
 		object.LastUpdateTimeUnix = now
 	}
-	class, err := b.schemaManager.GetClass(ctx, principal, object.Class)
-	ec.Add(err)
-	if class == nil {
-		ec.Add(fmt.Errorf("class '%s' not present in schema", object.Class))
-	} else {
-		err = validation.New(b.vectorRepo.Exists, b.config, repl).
-			Object(ctx, class, object, nil)
-		ec.Add(err)
 
-		if err == nil {
-			// update vector only if we passed validation
-			err = b.modulesProvider.UpdateVector(ctx, object, class, nil, b.findObject, b.logger)
-			ec.Add(err)
+	// Batch together the GetClass and the validation function as the validation function will create/update the class
+	// if it already exists to match the object.
+	err = backoff.Retry(func() error {
+		class, err := b.schemaManager.GetClass(ctx, principal, object.Class)
+		if err != nil {
+			return err
 		}
-	}
+
+		if class == nil {
+			return fmt.Errorf("class '%s' not present in schema", object.Class)
+		}
+
+		err = validation.New(b.vectorRepo.Exists, b.config, repl).Object(ctx, class, object, nil)
+		if err != nil {
+			return err
+		}
+
+		// update vector only if we passed validation
+		err = b.modulesProvider.UpdateVector(ctx, object, class, nil, b.findObject, b.logger)
+		if err != nil {
+			return err
+		}
+
+		return nil
```

Comment:
we can use here `return backoff.Permanent(err)` and it will make that it gonna retry every time until MAX retry matches the count

Reply:
Here we have no error to return however, it's a success. On all the other places where we return an error we might want to retry. So I don't think using Permanent would be beneficial. The doc states 

> If o returns a *PermanentError, the operation is not retried, and the wrapped error is returned. 

So for me it sounds like a terminal "don't retry further", and it's not what we want (if we already return nil it's the same).